### ISSUE TITLE
Ability to invoke async JavaScript functions and methods

### ISSUE THREAD

**Issue by bricelam** [unoplatform/uno#3084]
## What would you like to be added:

I'd love to see something similar to Blazor's [IJSRuntime.InvokeAsync](https://docs.microsoft.com/aspnet/core/blazor/call-javascript-from-dotnet) method. The workaround suggested in #65 is less than ideal.

## Why is this needed:

I'm trying to leverage the [oidc-client](https://github.com/IdentityModel/oidc-client-js) JavaScript library whose methods are all async (return Promise).

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

**Comment by bricelam** [comment]
I was able to leverage mono-wasm for now.

```xml
<ItemGroup>
  <Reference Include="WebAssembly.Bindings">
    <!-- TODO: Update this whenever Uno.Wasm.Bootstrap does -->
    <HintPath>$(TEMP)\mono-wasm-34cccfa2fde\framework\WebAssembly.Bindings.dll</HintPath>
    <Private>false</Private>
  </Reference>
</ItemGroup>
```

```cs
var oidcModule = (JSObject)Runtime.GetGlobalObject("Oidc");            
var userManagerType = (Function)oidcModule.GetObjectProperty("UserManager");
var userManager = Runtime.NewJSObject(userManagerType);

var user = (JSObject)await (Task<object>)userManager.Invoke("signinSilent");
```

**Comment by jeromelaban** [comment]
Indeed that's a way to use async methods there. The distribution model of the Bindings is not appropriate, and is very dependent on the mono version.

We'll most probably adjust that within the .NET 5 timeframe.

**Comment by carldebilly** [comment]
@bricelam This feature has been merged into `master` and available as version `3.0.0-dev.25` on [nuget](https://www.nuget.org/packages/Uno.UI/3.0.0-dev.25).
